Issue275 - Refactored Achievements Page

vanishajr opened the pull request:
## Summary:  What does this PR do?

Updated the entire frontend of the Achievements page. Added the AchievementsCategory model to change the structure of mongodb data which stores the achievements, hence new collection achievements_category is created which stores the new data. Added a new route to handle rendering of the achievements. Added CRUD operations as well.
CRUD now fetches achievements using the persons name. 

## Which issue(s) this PR fixes:

Fixes #275 

## Changes Made
Describe the changes you've made in this PR:
- Changed the frontend of achievements page.tsx
- Added a new route.ts file 
- Added AchievementsCategory.ts mongodb model

Changes can be verified on:
https://achievementspb.netlify.app/achievements

## Screenshots (if applicable)

https://github.com/user-attachments/assets/8ec8500a-161d-4483-a8f2-2254709f99ad

mohit-nagaraj commented:
minor nit: cards come up before the filters. If possible could u reduce the delay?

mohit-nagaraj commented on .github/workflows/pr-checks.yml:
could you discard the changes from this file?

mohit-nagaraj commented on package.json:
```diff
@@ -27,6 +27,7 @@
     "lodash": "^4.17.21",
     "lucide-react": "^0.419.0",
     "mongodb": "^6.15.0",
+    "mongodb": "^6.15.0",
```
mongodb 2 times? did something get messed up when merging?

vanishajr replied:
Yes, seems like an issue when merging, I've fixed it in the latest commit

mohit-nagaraj commented on package.json:
```diff
@@ -49,10 +50,14 @@
     "eslint-config-next": "14.2.13",
     "tailwindcss": "^3.3.6"
   },
+  "overrides": {
```
same [1]

vanishajr replied:
Fixed

mohit-nagaraj commented on package.json:
```diff
@@ -49,10 +50,14 @@
     "@tailwindcss/forms": "^0.5.7",
     "@types/aos": "^3.0.7",
     "@types/lodash": "^4.17.16",
+    "@types/lodash": "^4.17.16",
```
same [1]

vanishajr replied:
Fixed

mohit-nagaraj commented on components/achievementList.tsx:
```diff
@@ -152,233 +152,233 @@
       username: "@Innerve 9.0",
       body: "Winner of Innerve 9.0 Hackathon",
-      img: "innerve.png",
+      img: "innerve.webp",
```
what is the use of the 'img' field if u are not using it? this the new design doesnt require it, it would be a good option to drop this as well as the images if not being used anywhere else! Pls verify once with grep

vanishajr replied:
The code uses this file to render the achievements on the main page as the scrolling bar, hence I reverted the changes of this file, as my issue does not deal with the same code

mohit-nagaraj commented on app/(default)/api/achievements-category/route.ts:
```diff
@@ -0,0 +1,357 @@
+// POST method: Create or add a new achievement
+export async function POST(request: Request) {
+  try {
+    if (request.method !== 'POST') {
```
pls remove this is completely uncessary :)
the method u defined above is post so only those will hit here so this check would be false always

vanishajr replied:
Fixed

mohit-nagaraj commented on app/(default)/api/achievements-category/route.ts:
```diff
@@ -0,0 +1,357 @@
+// PUT method: Update an existing achievement based on name
+export async function PUT(request: Request) {
+  try {
+    if (request.method !== 'PUT') {
```
same [2]

vanishajr replied:
Fixed

mohit-nagaraj commented on app/(default)/api/achievements-category/route.ts:
```diff
@@ -0,0 +1,357 @@
+// GET method: Fetch achievements based on name or fetch all if no name is provided
+export async function GET(request: NextRequest) {
+  try {
+    if (request.method !== 'GET') {
```
same[2]

vanishajr replied:
Fixed

mohit-nagaraj commented on app/(default)/achievements/page.tsx:
```diff
@@ -56,677 +69,719 @@ export default function AchievementsPage() {
-      achievements: [...(prev.achievements || []), ""],
-    }));
+  const handleChangeAchievement = (category: string, index: number, field: keyof Achievement, value: string) => {
+    setNewAchievement(prev => {
```
could you once confirm me if this is working as intended thanks

mohit-nagaraj commented:
also @2004yash could you once take this pr n check locally if the achievements adding/updating works? Thanks

2004yash commented:
> also @2004yash could you once take this pr n check locally if the achievements adding/updating works? Thanks

Everything is working fine, just that when entering name, it can be a search kind of thing, like a list comes and we have to select name from the list. @vanishajr

vanishajr commented:
> > also @2004yash could you once take this pr n check locally if the achievements adding/updating works? Thanks
> 
> Everything is working fine, just that when entering name, it can be a search kind of thing, like a list comes and we have to select name from the list. @vanishajr

Added a dropdown list to select names or enter names manually while editing the achievements

vanishajr commented:
> minor nit: cards come up before the filters. If possible could u reduce the delay? also as pointed could you update/add placeholder for profile images

Fixed the animation so that cards appear after the filters and also added whileInView so that cards animate on scroll

mohit-nagaraj reviewed (approved):
lgtm! Great efforts ... thanks for the contribution
